develop/application-devs/debugging-errors.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Errors and How to Resolve Them
+description: Common errors that are encountered across the runtime, tooling, and block explorers.
+---
+
+Errors in Substrate-based chains are usually accompanied by descriptive messages. However, to read
+these messages, a tool parsing the blockchain data needs to request _chain metadata_ from a node.
+That metadata explains how to read the messages. One such tool with a built-in parser for chain
+metadata is the [Polkadot-JS Apps UI](https://polkadot.js.org/apps){target=\_blank}.
+
+If this page doesn't answer your question, try searching for your problem at the
+[Polkadot Knowledge Base](https://support.polkadot.network/){target=\_blank} for more information on troubleshooting
+your issue.
+
+## PolkadotJS Apps Explorer
+
+Here's how to find out the detailed error description through Polkadot-JS Apps.
+
+A failed transaction typically looks similar like this:
+
+![Error while sending a transaction](/images/applications-devs/errors/01.webp)
+
+The image displays only the error name as defined in the code, not its error message.
+
+In the [explorer tab](https://polkadot.js.org/apps/#/explorer){target=\_blank}, find the block in which this failure
+occurred. Then, expand the `system.ExtrinsicFailed` frame:
+
+![Error described](/images/applications-devs/errors/02.webp)
+
+Notice how the `details` field contains a human-readable description of the error. Most errors will
+have this, if looked up this way.
+
+[This block](https://polkadot.js.org/apps/#/explorer/query/0xa10104ed21dfe409c7871a975155766c5dd97e1e2ac7faf3c90f1f8dca89104b){target=\_blank} is a live example of this scenario.
+
+If the error can't be found, or there is no message in the `details` field, consult the table below.
+
+!!!info "Future Error"
+
+    This error will not cause the transaction to be discarded immediately. Instead, it will be sent to the future queue, where it
+    will wait to be executed at the correct place in the nonce sequence or it will get discarded due to
+    some other error (ex. the validity period expires).
+
+## Subscan
+
+The `ExtrinsicFailed` event indicates when a transaction doesn't succeed ([example](https://polkadot.subscan.io/extrinsic/19983878-2?event=19983878-53){target=\_blank}). This event provides the `error` and `index` (as seen in the table of the event, in the `dispatch_error` row) indices of the error but doesn't provide an informative message to understand what it means. The error can be identified in the codebase to understand what went wrong.
```

Comment:
Explain what subscan is or at least add a link to it. It is the first time being introduced. 

An example could be added showing the 'error' and 'index', and then a snippet with the source code matching that error